### PR TITLE
bwa v. 0.6.2!

### PR DESCRIPTION
**Pull request by kthlnktng**

### CODE REVIEW

**Comment by kthlnktng** [comment]
# For future reference:

Unlike many of the other previously generated spec files (e.g. [gcc-6.2.0](o2/gcc-6.2.0.spec)), `bwa`'s [recommended build process](https://github.com/lh3/bwa#getting-started) does not have a `./configure` step. This means I couldn't specify `--prefix=%{INSTALL_ROOT}/%{name}/%{version}`, which is why I was running into all of these errors about being in the wrong directory (for all jenkins builds I started before [59](http://jenkins.rc.hms.harvard.edu/job/test_deploy_applicationstack/59/) / 958912ece5cfd282ddbb6c65fb833724fd0f05ae commit).

__This is the general process of what is happening in what directory in the testing spec file__:
1. `make` is run in `/n/app/rc-testing/rpmbuild/BUILD/bwa-0.6.2`
2. Directory `/n/app/rc-testing/apps-test/bwa/0.6.2` aka `%{INSTALL_ROOT}/%{name}/%{version}` is created. Files from  `/n/app/rc-testing/rpmbuild/BUILD/bwa-0.6.2` are copied to `/n/app/rc-testing/apps-test/bwa/0.6.2`.
3. *required for %files to work*: files from `/n/app/rc-testing/apps-test/bwa/0.6.2` aka `%{INSTALL_ROOT}/%{name}/%{version}` are copied to `/n/app/rc-testing/rpmbuild/BUILDROOT/bwa-0.6.2-1.el7.centos.x86_64/0.6.2` aka `$RPM_BUILD_ROOT/%{version}`

**Comment by kthlnktng** [comment]
Sure, will update with those changes.

File: o2/bwa-0.6.2-generic.spec [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+%files
+%defattr(775,jenkins-slave,ritg,775)
+/%{version}/*
+
+# make relevant notations here IF it completely affects a rebuild of the application
+# most recent changes are on top; date takes the form of `date +"%a %b %d %Y"`
+# if a change here modifies the overall function of the application (as opposed to a bugfix/path change, etc.), bump the Release version (e.g. 1 -> 2)
+# and note it in the changelog
+%changelog
+* Tue Nov 29 2016 Kathleen Keating <kathleen_keating@hms.harvard.edu>
```
Comment (alextruong) [inline o2/bwa-0.6.2-generic.spec]: Nov 30

Comment (kthlnktng) [inline o2/bwa-0.6.2-generic.spec]: Oops.

File: o2/bwa-0.6.2-generic.spec [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# rename all source tarballs/archives to look like this (changing archive types is fine, e.g. .tgz, .zip, etc.)
+Source0: %{name}-%{version}.tar.gz
+# The location where Source0 was downloaded - if it was renamed to look like Source0, PROVIDE ORIGINAL NAME HERE
+#Downloaded from: https://github.com/lh3/bwa/archive/0.6.2.tar.gz
+
+%description
+BWA is a software package for mapping DNA sequences against a large reference genome, such as the human genome. It consists of three algorithms: BWA-backtrack, BWA-SW and BWA-MEM. The first algorithm is designed for Illumina sequence reads up to 100bp, while the rest two for longer sequences ranged from 70bp to a few megabases. BWA-MEM and BWA-SW share similar features such as the support of long reads and chimeric alignment, but BWA-MEM, which is the latest, is generally recommended as it is faster and more accurate. BWA-MEM also has better performance than BWA-backtrack for 70-100bp Illumina reads.
+
+# change this if debugging (e.g. installing locally somewhere)
+%define INSTALL_ROOT /n/app/rc-testing/apps-test
```
Comment (alextruong) [inline o2/bwa-0.6.2-generic.spec]: change path to production

**Pull request merged**